lib/topological_inventory/providers/common/operations/async_worker.rb

```diff
@@ -0,0 +1,42 @@
+require "topological_inventory/providers/common/logging"
+require "topological_inventory/providers/common/operations/health_check"
+
+module TopologicalInventory
+  module Providers
+    module Common
+      module Operations
+        class AsyncWorker
+          include Logging
+
+          def initialize(queue, impl_class)
+            @queue = queue
+            @impl = impl_class
```

Comment:
I would prefer some better name, it't not obvious what `impl` means

Reply:
I just changed it to `processor` since that is the class we'll be passing into it most every time for the workers.